tsdb/db.go

```diff
@@ -761,6 +765,10 @@ func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs
 		stats = NewDBStats()
 	}
 
+	if opts.PostingsDecoderFactory == nil {
+		opts.PostingsDecoderFactory = DefaultPostingsDecoderFactory
```

Comment:
I don't think you should alter the options passed in.

Reply:
Removed